eZ/Publish/Core/Persistence/Legacy/Content/Gateway/ExceptionConversion.php

```diff
@@ -886,4 +886,27 @@ public function insertRelation( RelationCreateStruct $struct )
             throw new RuntimeException( 'Database error', 0, $e );
         }
     }
+
+    /**
+     * Returns all Content IDs for a given $contentTypeId.
+     *
+     * @param int $contentTypeId
+     *
+     * @return int[]
+     */
+    public function getContentIdsByType( $contentTypeId )
```

Comment:
Not with the code here but we Have contenTypeId and ContentTypeIdentifier search criterions, so...
Maybe someone think ByContenTtype needs a ContentType object Instead of an id. Obviously rest of the code will need to be updated.